### PR TITLE
Tabs and Accordion: Account for Accented Characters

### PR DESCRIPTION
**Pull request by AlexGStapleton**
This is a follow up to #794. This PR uses the provided feedback and also accounts for accented characters for the accordion widget also.

[Accordion Test Layout](https://drive.google.com/a/siteorigin.com/uc?id=1J7Avs2zold9rSt6032r0Kb61ZL1bkJwq)
[Tab Test Layout](https://drive.google.com/a/siteorigin.com/uc?id=1pggjX64KJ6LAIrohlCGD3w6222_i090a)

### CODE REVIEW

**Comment by braamgenis** [comment]
I still can't reproduce, even with the layouts you provided. :finnadie:  I assume this has been reported again? If so, it would be great to see the thread for more info. Also, as [I mentioned](https://github.com/siteorigin/so-widgets-bundle/pull/794#issuecomment-450594946) in the previous PR, it should be fine to use `decodeURI` but let's not change anything else.

**Comment by AlexGStapleton** [comment]
I'm not too sure why you're unable to replicate this.
How does that look?

File: widgets/tabs/js/tabs.js [hunk 1]
```diff
@@ -121,7 +121,9 @@ jQuery( function ( $ ) {
 					if ( window.location.hash ) {
 						var anchors = window.location.hash.replace( '#', '' ).split( ',' );
 						anchors.forEach( function ( anchor ) {
-							var tab = $tabs.filter( '[data-anchor="' + anchor + '"]' );
+							var tab = $tabs.filter( function ( index, element ) {
+								return decodeURI( anchor ) === decodeURI( $( element ).data( 'anchor' ) );
+							} );
 							if ( tab ) {
```
Comment (braamgenis) [inline widgets/tabs/js/tabs.js]: The return value of the jQuery filter method is a jQuery object, so `tab` will always be truthy. A more accurate test here is perhaps `if ( tab.length > 0 )`.

File: widgets/accordion/js/accordion.js [hunk 2]
```diff
@@ -142,7 +142,12 @@ jQuery( function ( $ ) {
 						var panel = panels[ i ];
 						var anchor = $( panel ).data( 'anchor' );
 						var anchors = window.location.hash.substring(1).split( ',' ); 
-						if ( anchor && $.inArray( anchor.toString(), anchors ) > -1 ) {
+						
+						var anchor = anchors.filter( function ( index, element ) {
+							return decodeURI( anchor.toString() ) === decodeURI( index );
```
Comment (braamgenis) [inline widgets/accordion/js/accordion.js]: This is decoding the integer index of the element and comparing it to the decoded `anchor` string . It should probably be something more like `return decodeURI( anchor.toString() ) === decodeURI( $( element ).data( 'anchor' ) );`.

File: widgets/accordion/js/accordion.js [hunk 3]
```diff
@@ -142,7 +142,12 @@ jQuery( function ( $ ) {
...
+						} );
+
+						if ( anchor.length ) {
```
Comment (braamgenis) [inline widgets/accordion/js/accordion.js]:
```suggestion
						if ( panelOpen ) {
```

File: widgets/accordion/js/accordion.js [hunk 4]
```diff
@@ -142,7 +142,12 @@ jQuery( function ( $ ) {
 						var panel = panels[ i ];
 						var anchor = $( panel ).data( 'anchor' );
```
Comment (braamgenis) [inline widgets/accordion/js/accordion.js]:
```suggestion
						var panelAnchor = $( panel ).data( 'anchor' );
```

File: widgets/accordion/js/accordion.js [hunk 5]
```diff
@@ -142,7 +142,12 @@ jQuery( function ( $ ) {
...
 						var anchors = window.location.hash.substring(1).split( ',' ); 
-						if ( anchor && $.inArray( anchor.toString(), anchors ) > -1 ) {
+						
+						var anchor = anchors.filter( function ( index, element ) {
```
Comment (braamgenis) [inline widgets/accordion/js/accordion.js]:
```suggestion
						var panelOpen = anchors.some(function (anchor) {
```

File: widgets/accordion/js/accordion.js (hunk 2 above)
Comment (braamgenis) [inline widgets/accordion/js/accordion.js]:
```suggestion
							return decodeURI(anchor) === decodeURI(panelAnchor);
```

File: widgets/tabs/js/tabs.js (hunk 1 above)
Comment (braamgenis) [inline widgets/tabs/js/tabs.js]:
```suggestion
							if ( tab.length > 0 ) {
```

**Pull request closed without merging**